powerline/segments/i3wm.py

```diff
@@ -39,27 +48,58 @@ def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
 
 	Highlight groups used: ``workspace`` or ``w_visible``, ``workspace`` or ``w_focused``, ``workspace`` or ``w_urgent``.
 	'''
-	global conn
-	if not conn:
-		try:
-			import i3ipc
-		except ImportError:
-			import i3 as conn
-		else:
-			conn = i3ipc.Connection()
-
 	output = output or segment_info.get('output')
 
 	return [{
-		'contents': w['name'][min(len(w['name']), strip):],
+		'contents': w['name'][strip:],
 		'highlight_groups': calcgrp(w)
-	} for w in conn.get_workspaces()
-		if (not only_show or any(w[typ] for typ in only_show))
-		and (not output or w['output'] == output)
+	} for w in get_i3_connection().get_workspaces()
+		if ((not only_show or any(w[typ] for typ in only_show))
+		and (not output or w['output'] == output))
 	]
 
 
 @requires_segment_info
+def workspace(pl, segment_info, workspace=None, strip=False):
+	'''Return state for the specified workspace
```

Comment:
> actually they differ:
> 
>     def workspaces(pl, segment_info, only_show=None, output=None, strip=0):
>         	'''Return list of used workspaces
> 
> vs
> 
>     def workspace(pl, segment_info, workspace=None, strip=False):
>         	'''Return state for the specified workspace

I commented near `:param bool strip:` because it is what had the problem. But now that you highlighted it I see that documentation is not valid: `workspace` segment does not return *state* it returns workspace *name*. Usage of different highlight group is irrelevant and should not go into the summary: see how this is described in `powerline.segments.common.env.user`.